src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -21,92 +44,1623 @@ const ActorInit En_In_InitVars = {
+                    func_800E8EA0(globalCtx, &this->actor, 0x3465);
+                    ret = 0;
+                    break;
+                case 0x3465:
+                    func_800E8EA0(globalCtx, &this->actor, 0x3466);
+                    ret = 0;
+                    break;
+                case 0x3466:
+                    if (msgCtx->choiceIndex == 0) {
+                        func_8019F208();
+                        if (gSaveContext.rupees >= globalCtx->msgCtx.unk1206C) {
+                            if (func_80114E90()) {
+                                this->actionFunc = func_808F3C40;
+                                func_800B8A1C(&this->actor, globalCtx, 0x92, 500.0f, 100.0f);
+                                func_801159EC(-globalCtx->msgCtx.unk1206C);
+                                ret = 1;
+                            } else {
+                                func_800E8EA0(globalCtx, &this->actor, 0x3469);
+                                ret = 0;
+                            }
+                        } else {
+                            play_sound(0x4806);
+                            func_800E8EA0(globalCtx, &this->actor, 0x3468);
+                            ret = 0;
+                        }
+                    } else {
+                        func_8019F230();
+                        func_800E8EA0(globalCtx, &this->actor, 0x3467);
+                        ret = 0;
+                    }
+                    break;
+                case 0x3467:
+                case 0x3468:
+                case 0x3469:
+                    func_80151BB4(globalCtx, 0x11);
+                    ret = 1;
+                    break;
+                case 0x346B:
+                    func_800E8EA0(globalCtx, &this->actor, 0x346C);
+                    ret = 0;
+                    break;
+                case 0x346C:
+                    func_800E8EA0(globalCtx, &this->actor, 0x3466);
+                    ret = 0;
+                    break;
+                case 0x346A:
+                    this->actionFunc = func_808F5A94;
+                    func_80151BB4(globalCtx, 0x11);
+                    break;
+            }
+            break;
+        case 2:
+            switch (textId) {
+                case 0x346E:
+                    func_808F4108(this, globalCtx, 0x346F);
+                    ret = 0;
+                    break;
+                case 0x3470:
+                    func_800E8EA0(globalCtx, &this->actor, 0x3471);
+                    ret = 0;
+                    break;
+                case 0x3483:
+                    func_800E8EA0(globalCtx, &this->actor, 0x3471);
+                    ret = 0;
+                    break;
+                case 0x3471:
+                    func_808F4150(globalCtx, this, arg2, msgCtx);
+                    ret = 0;
+                    break;
+                case 0x3472:
+                    func_808F43E0(this);
+                    gSaveContext.weekEventReg[56] &= (u8)~8;
```

Comment:
Is the cast really needed?
```suggestion
                    gSaveContext.weekEventReg[56] &= ~8;
```

Reply:
The cast is needed for an OK. Without it, IDO ands with 0xFFF7 instead of 0xF7.